includes/Classifai/Helpers.php

```diff
@@ -302,3 +302,33 @@ function get_feature_taxonomy( $feature ) {
 	 */
 	return apply_filters( 'classifai_taxonomy_for_feature', $taxonomy, $feature );
 }
+
+/**
+ * Filter the SQL clauses of an attachment query to include tags and alt text.
+ *
+ * @param array $clauses An array including WHERE, GROUP BY, JOIN, ORDER BY,
+ *                       DISTINCT, fields (SELECT), and LIMITS clauses.
+ * @return array The modified clauses.
+ */
+function filter_attachment_query_keywords( array $clauses ) {
+	global $wpdb;
+	remove_filter( 'posts_clauses', __FUNCTION__ );
+
+	if ( ! preg_match( "/\({$wpdb->posts}.post_content (NOT LIKE|LIKE) (\'[^']+\')\)/", $clauses['where'] ) ) {
+		return $clauses;
+	}
+
+	// Add a LEFT JOIN of the postmeta table so we don't trample existing JOINs.
+	$clauses['join'] .= " LEFT JOIN {$wpdb->postmeta} AS classifai_postmeta ON ( {$wpdb->posts}.ID = classifai_postmeta.post_id AND ( classifai_postmeta.meta_key = 'classifai_computer_vision_image_tags' OR classifai_postmeta.meta_key = '_wp_attachment_image_alt' ) )";
+
+	$clauses['groupby'] = "{$wpdb->posts}.ID";
+
+	$clauses['where'] = preg_replace(
+		"/\({$wpdb->posts}.post_content (NOT LIKE|LIKE) (\'[^']+\')\)/",
+		'$0 OR ( classifai_postmeta.meta_value $1 $2 )',
+		$clauses['where']
+	);
+
+	return $clauses;
+}
+\add_filter( 'posts_clauses', __NAMESPACE__ . '\\filter_attachment_query_keywords' );
```

Comment:
Do we need the extra `\` before the function name? I feel like we don't but I might just be under-caffeinated.

Reply:
No, I don’t think so, I was just trying to see if it would fix the linting error. Any ideas there @ryanwelcher? I don’t know why the file is autoloaded like that, I guess I could move this into a different file if that makes more sense.